### PR TITLE
Store centric UI: Part 3

### PR DESCRIPTION
**Pull request by dennisreimann**
* Store context is kept for Account and Server Settings 
* Store switch links to wallet transactions if wallet is set up - otherwise it links to the payment methods, where the wallet can be set up.
* Integrations now appear in the nav under Plugins
* Pay Button got moved to the sidebar nav under Apps
* Correct nav item highlighting

### CODE REVIEW

File: BTCPayServer/Security/CookieAuthorizationHandler.cs [hunk 1]
```diff
@@ -108,17 +105,27 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
                     if (store != null || isAdmin)
                         success = true;
                     break;
+                case Policies.CanViewProfile:
+                    // FIXME: Is there a better way to check this?
+                    var endpoint = context.Resource as RouteEndpoint;
+                    if (endpoint?.RoutePattern?.RawText.IndexOf("Manage/", StringComparison.InvariantCultureIgnoreCase) == 0 || isAdmin)
```
Comment (dennisreimann) [inline BTCPayServer/Security/CookieAuthorizationHandler.cs:111]: This check is weird and potentially unnecessary, but I wanted to ensure we are dealing with the account settings and not the settings of another user.

Comment (NicolasDorier) [reply BTCPayServer/Security/CookieAuthorizationHandler.cs:114]: I don't understand why you need to add this policy?
Why would admin have access to view of all profiles?

Comment (dennisreimann) [reply BTCPayServer/Security/CookieAuthorizationHandler.cs:114]: Ok, I'll simply remove it then.

**Review by Kukks — CHANGES_REQUESTED** [review]

File: BTCPayServer/Views/Wallets/_Nav.cshtml [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
-    var network = BtcPayNetworkProvider.GetNetwork<BTCPayNetwork>(wallet.CryptoCode);
+    var walletId = Context.GetRouteValue("walletId")?.ToString();
+    var storeId = Context.GetRouteValue("storeId")?.ToString();
+    var cryptoCode = Context.GetRouteValue("cryptoCode")?.ToString();
+    var wallet = walletId != null ? WalletId.Parse(walletId) : new WalletId(storeId, cryptoCode);
+    var network = _btcPayNetworkProvider.GetNetwork<BTCPayNetwork>(wallet.CryptoCode);
 }
 <nav id="SectionNav" class="nav">
-    <a class="nav-link @ViewData.IsActivePage(WalletsNavPages.Transactions)" asp-action="WalletTransactions" asp-route-walletId="@Context.GetRouteValue("walletId")" id="SectionNav-Transactions">Transactions</a>
+    <a class="nav-link @ViewData.IsActivePage(WalletsNavPages.Transactions)" asp-controller="Wallets" asp-action="WalletTransactions" asp-route-walletId="@wallet" id="SectionNav-Transactions">Transactions</a>
```
[the remark this answers is not in the document]
Comment (dennisreimann) [reply BTCPayServer/Views/Wallets/_Nav.cshtml]: Done in c9f0b81.

**Comment by dennisreimann** [comment]
Now all user-facing actions should have policies and hence the store context. 

This PR is ready for review now. Will send a separate last one once this is merged.

File: BTCPayServer/Controllers/VaultController.cs [hunk 3]
```diff
@@ -18,6 +19,7 @@
 namespace BTCPayServer.Controllers
 {
     [Route("vault")]
+    [Authorize(AuthenticationSchemes = AuthenticationSchemes.Cookie, Policy = Policies.CanModifyStoreSettings)]
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/VaultController.cs:22]: why? not needed afaik

File: BTCPayServer/Controllers/UserStoresController.cs [hunk 4]
```diff
@@ -14,40 +15,36 @@
 namespace BTCPayServer.Controllers
 {
     [Route("stores")]
-    [Authorize(AuthenticationSchemes = AuthenticationSchemes.Cookie)]
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/UserStoresController.cs:17]: why did you remove it from the class level down to action level?

File: BTCPayServer/Security/CookieAuthorizationHandler.cs [hunk 5]
```diff
@@ -81,52 +85,107 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
...
                     if (store != null && (store.Role == StoreRoles.Owner || isAdmin))
                         success = true;
                     break;
                 case Policies.CanViewStoreSettings:
+                    store = await storeT.Value;
                     if (store != null || isAdmin)
                         success = true;
                     break;
+                case Policies.CanViewInvoices:
+                    success = true;
```
Comment (NicolasDorier) [inline BTCPayServer/Security/CookieAuthorizationHandler.cs:143]: I don't understand this, it means that everybody has CanViewInvoices permissions over every store.

File: BTCPayServer/Security/CookieAuthorizationHandler.cs [hunk 6]
```diff
@@ -81,52 +85,107 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
...
+                    if (usedCookieFallback)
```
Comment (NicolasDorier) [inline BTCPayServer/Security/CookieAuthorizationHandler.cs:144]: I don't understand this.

Comment (dennisreimann) [reply BTCPayServer/Security/CookieAuthorizationHandler.cs]: Simplified it in 2e7d830.

File: BTCPayServer/Controllers/UserStoresController.cs (hunk 4 above)
Comment (dennisreimann) [reply BTCPayServer/Controllers/UserStoresController.cs:17]: Fixed in 97e8b83.

File: BTCPayServer/Controllers/VaultController.cs (hunk 3 above)
Comment (dennisreimann) [reply BTCPayServer/Controllers/VaultController.cs]: Fixed in 97e8b83.

**Comment by dennisreimann** [comment]
@kukks Please also take a look at 2e7d830: I could simplify it by distinguishing the global and store case on the  [controller level](https://github.com/btcpayserver/btcpayserver/pull/3224/files#diff-06fcf681839e3a4010482db3e64aba26d0e6cab7d207514f7150d8976b8355e4R750). 

So it is either:
- `/invoices/` (unfiltered list)
- `/invoices?SearchTerm=storeId%3AID` (filters that store on the global list without settting the store id cookie value; there can also be multiple `storeId` filters)
- `/stores/ID/invoices` just invoices for that particular store, sets the store id cookie value

I think it is best to approach it like this, because it gives us easy options to handle the different cases of which we can potentially eliminate the global invoices one in the future.

**Pull request merged**